### PR TITLE
Add spring-cloud-azure-native-configuration build customizer

### PR DESCRIPTION
**Pull request by moarychan**
The artifact id `spring-cloud-azure-native-configuration` will be added automatically when there's any Spring Cloud Azure coordinate that supports Spring Native.

### CODE REVIEW

File: start-site/src/main/resources/application.yml [hunk 1]
```diff
@@ -281,6 +281,25 @@ initializr:
             - rel: reference
               href: https://docs.spring.io/spring-native/docs/current/reference/htmlsingle/
               description: Spring Native Reference Guide
+        - name: Spring Cloud Azure Native Configuration
+          id: azure-native-configuration
```
Comment (backwind1233) [inline start-site/src/main/resources/application.yml:285]: add bom section here

File: initializr-extension/src/main/java/com/azure/spring/initializr/extension/dependency/springazure/AbstractSpringCloudAzureNativeBuildCustomizer.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        if (supportValidator.unSupport(build)) {
+            return;
+        }
+
+        Dependency selected = build.dependencies().items()
+                                   .filter(u -> u.getGroupId().equals("com.azure.spring"))
+                                   .findAny()
+                                   .get();
+        logger.info("Select spring.cloud-azure.version: " + selected.getVersion() + ".");
+        Dependency.Builder<?> builder = withCoordinates("com.azure.spring", "spring-cloud-azure-native-configuration");
```
Comment (backwind1233) [inline initializr-extension/src/main/java/com/azure/spring/initializr/extension/dependency/springazure/AbstractSpringCloudAzureNativeBuildCustomizer.java:61]: can we just use 
```
build.dependencies()
             .add("azure-native-configuration") 
```
?

Comment (moarychan) [reply initializr-extension/src/main/java/com/azure/spring/initializr/extension/dependency/springazure/AbstractSpringCloudAzureNativeBuildCustomizer.java:55]: OK

File: start-site/src/main/resources/application.yml (hunk 1 above)
Comment (moarychan) [reply start-site/src/main/resources/application.yml:285]: It's a special case that is not managed in spring-cloud-azure-dependencies bom.

Comment (backwind1233) [reply start-site/src/main/resources/application.yml:285]: Ok, I see

File: initializr-extension/src/main/java/com/azure/spring/initializr/autoconfigure/ExtendDependencyProjectGenerationConfiguration.java [hunk 3]
```diff
@@ -44,18 +51,34 @@ public SpringAzureDefaultBuildCustomizer springAzureBuildCustomizer() {
...
 	}
 
 	@Bean
+    @ConditionalOnRequestedDependency("cloud-stream")
 	public SpringAzureMessagingBuildCustomizer springAzureMessagingBuildCustomizer() {
 		return new SpringAzureMessagingBuildCustomizer();
 	}
 
 	@Bean
+    @ConditionalOnRequestedDependency("cloud-starter-sleuth")
```
Comment (saragluna) [inline initializr-extension/src/main/java/com/azure/spring/initializr/autoconfigure/ExtendDependencyProjectGenerationConfiguration.java:66]: These show an inconsistent indentation.

File: initializr-extension/src/main/java/com/azure/spring/initializr/extension/dependency/springazure/AbstractSpringCloudAzureNativeBuildCustomizer.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import static com.azure.spring.initializr.extension.dependency.springazure.SpringCloudAzureNativeConfigurationVersionResolver.resolve;
+import static io.spring.initializr.generator.buildsystem.Dependency.withCoordinates;
+
+/**
+ * An abstract {@link BuildCustomizer} class that automatically adds "com.azure.spring:spring-cloud-azure-native-configuration" when relevant Spring Cloud Azure
+ * library and Spring Native are selected.
+ */
+abstract class AbstractSpringCloudAzureNativeBuildCustomizer<T extends Build> implements BuildCustomizer<T>, Ordered {
+
+    private static final Log logger = LogFactory.getLog(AbstractSpringCloudAzureNativeBuildCustomizer.class);
```
Comment (saragluna) [inline initializr-extension/src/main/java/com/azure/spring/initializr/extension/dependency/springazure/AbstractSpringCloudAzureNativeBuildCustomizer.java:38]: Seems like in this PR we are using `apache.commons.logging`, but in another PR we are using SLF4J.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: initializr-extension/src/main/java/com/azure/spring/initializr/extension/dependency/springazure/AbstractSpringCloudAzureNativeBuildCustomizer.java (hunk 4 above)
Comment (moarychan) [reply initializr-extension/src/main/java/com/azure/spring/initializr/extension/dependency/springazure/AbstractSpringCloudAzureNativeBuildCustomizer.java:38]: https://docs.spring.io/spring-framework/docs/5.0.0.M5/spring-framework-reference/html/overview.html#overview-not-using-commons-logging, the log should be unified use of slf4j API.

**Comment by stliu** [comment]
drop from plan

**Pull request closed without merging**